### PR TITLE
Like comment option

### PR DESCRIPTION
**Pull request by sserbula**

### CODE REVIEW

File: Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs [hunk 1]
```diff
@@ -23,6 +24,41 @@ public CommentCommandRepository(IMealOrderingContextFactory factory)
...
+                        if (comment.CustomerId != entity.CustomerId && reaction != null)
+                        {
+                            context.ReactionToComment.Remove(reaction);
+                            context.SaveChanges();
+
+                            return new Right<Error, bool>(true);
+                        }
+                    }
+
+                    return new Left<Error, bool>(new RecordNotFound(nameof(entity)));
```
Comment (nikolicdusan) [inline Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs:53]:
```suggestion
                    var comment = context.Comment.Find(entity.MealId, entity.CommentId);
                    if (comment != null && comment.CustomerId == entity.CustomerId)
                    {
                        var reaction = context.ReactionToComment.SingleOrDefault(rc => rc.MealId == comment.MealId
                                                        && rc.CommentId == comment.CommentId
                                                        && rc.CustomerId == entity.CustomerId);
                        if (reaction != null)
                        {
                            context.ReactionToComment.Remove(reaction);
                            context.SaveChanges();

                            return new Right<Error, bool>(true);
                        }
                    }

                    return new Left<Error, bool>(new RecordNotFound(nameof(entity)));
```

File: Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs [hunk 2]
```diff
@@ -58,6 +94,42 @@ public CommentCommandRepository(IMealOrderingContextFactory factory)
...
+                {
+                    var comment = context.Comment.SingleOrDefault(c => c.MealId == entity.MealId && c.CommentId == entity.CommentId);
+                    if (comment != null && CheckIfCommentReactionCanBeSaved(context, comment, entity))
+                    {
+                        var commentReaction = new ReactionToCommentEntity
+                        {
+                            MealId = entity.MealId,
+                            CommentId = entity.CommentId,
+                            CustomerId = entity.CustomerId,
+                            ReactionId = (long)Reaction.Like
```
Comment (nikolicdusan) [inline Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs:116]: Every reaction is `Reaction.Like`?

File: Exebite.API/Controllers/CommentsController.cs [hunk 3]
```diff
@@ -42,6 +42,24 @@ public class CommentsController : ControllerBase
...
+        [HttpPost("reaction")]
+        [Authorize(Policy = nameof(AccessPolicy.CreateCommentReactionAccessPolicy))]
+        public IActionResult PostReaction([FromBody] InsertCommentReactionDto commentReaction) =>
+            _mapper.Map<CommentInsertReactionModel>(commentReaction)
+                   .Map(_commandRepository.InsertReaction)
+                   .Map(x => Created(new { saved = x }))
+                   .Reduce(_ => BadRequest(), error => error is ValidationError)
+                   .Reduce(_ => InternalServerError(), x => _logger.LogError(x.ToString()));
+
+        [HttpDelete("reaction")]
```
Comment (nikolicdusan) [inline Exebite.API/Controllers/CommentsController.cs:54]: It's common to provide the `{id}` of the record that should be deleted.

```suggestion
        [HttpDelete("reaction/{reactionId}")]
```

File: Test/Exebite.DataAccess.Test/CommentReactionRepositoryTest.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace Exebite.DataAccess.Test
+{
+    public class CommentReactionRepositoryTest
+    {
+    }
+}
```
Comment (nikolicdusan) [inline Test/Exebite.DataAccess.Test/CommentReactionRepositoryTest.cs:6]: Let's not push empty classes with no use whatsoever.

**Review by nikolicdusan — CHANGES_REQUESTED** [review]
Please address the review comments

File: Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs (hunk 2 above)
Comment (sserbula) [reply Exebite.DataAccess/Repositories/CommentRepository/CommentCommandRepository.cs:116]: We can only like comments and we already had table `Reaction` with `Like` entity, so I used it. When customer remove like, I remove entity from `ReactionToComment` table

**Review by nikolicdusan — CHANGES_REQUESTED** [review]
Review comments

**Review by nikolicdusan — APPROVED** [review]
Looks good

**Pull request merged**